### PR TITLE
Add some clarification to the async dispatch docs

### PR DESCRIPTION
**Pull request by yashk2810**

### CODE REVIEW

File: docs/async_dispatch.rst [hunk 1]
```diff
@@ -10,6 +10,8 @@ program:
 >>> import jax.numpy as jnp
 >>> from jax import random
 >>> x = random.uniform(random.PRNGKey(0), (1000, 1000))
+>>> # Printing the result or repr(result) or str(result) will block until the
```
Comment (mattjj) [inline docs/async_dispatch.rst:13]:
```suggestion
>>> # Printing the result (i.e. evaluating `repr(result)` or `str(result)`) will block until the
```

Maybe also tweak this to wrap at 80 character lines.

File: docs/async_dispatch.rst [hunk 2]
```diff
@@ -10,6 +10,8 @@ program:
...
+>>> # value is ready. That's why the value is present in the output below.
```
Comment (mattjj) [inline docs/async_dispatch.rst:14]:
```suggestion
>>> # value is ready.
```

I'm not sure but I suspect this second sentence isn't clear without more context.

**Review by mattjj — APPROVED** [review]
Thanks for the attention to detail! I suggested some tweaks; WDYT?

File: docs/async_dispatch.rst (hunk 1 above)
Comment (yashk2810) [reply docs/async_dispatch.rst]: Done

File: docs/async_dispatch.rst (hunk 2 above)
Comment (yashk2810) [reply docs/async_dispatch.rst]: Done

**Pull request merged**